lib/functions/transferpartsjoin.js

```diff
@@ -78,6 +78,11 @@ class JoinTransferParts extends AsyncGeneratorFunction {
                 // check if the transfer has been completed
                 const contentLength = transferPart.metadata.contentLength;
                 if (isComplete(completedRanges, contentLength)) {
+                    const startTime = controller.getTransferStartTime(transferAsset);
+                    if (startTime) {
+                        const uploadDuration = new Date().getTime() - startTime;
+                        transferAsset.setUploadDuration(uploadDuration);
+                    }
```

Comment:
This can then be `transferAsset.getStartTime()` and calculate the duration the same way.